Added section on Templated movies to Saving Movies doc. (#17793)
visit-dav/visit: src/doc/using_visit/SavingPrinting/Saving_movies.rst

   Movie progress dialog

The second movie generation method will cause VisIt_ to save out a session file containing every detail about your visualization so it can be recreated by a new instance of VisIt_.
This method works well if you want to create a movie without sacrificing your current VisIt_ session but you cannot watch the movie as it is generated and you may have to wait for the second instance's compute engine to be scheduled to run.
The last movie generation option simply makes VisIt_ display the command that you would have to type at a command prompt in order to make VisIt_ generate a movie of your current visualizations.

Templated movies
~~~~~~~~~~~~~~~~

Movie templates enable you to create a set of viewports and map the contents of various visualization windows to those viewports, also incorporating simple movie transitions.

To create a movie template, you can open the Save movie wizard as you normally would.
Instead of choosing New simple movie, choose New template movie.

> VisIt will load custom user interfaces pages for the chosen movie template.

Reviewer: Custom user interfaces pages? Or custom user interface pages? Or a custom user interfaces page?
I can't tell if it's supposed to be the way it currently is.

.. _savemovietemplate1:

.. figure:: images/savemovietemplate1.png 
   :scale: 75%
 
   Movie type selection page

You can choose to use an existing template, edit an existing template, or create a new template.

.. _savemovie_existingtemplate:

.. figure:: images/savemovie_existingtemplate.png 
   :scale: 75%
   
   Template choice.

Existing Movie Templates
""""""""""""""""""""""""

.. _CurveOverlayTemplate:

Curve Overlay Movie Template
^^^^^^^^^^^^^^^^^^^^^^^^^^^^

The **Curve Overlay** template lets you create regular plots and in a separate viewport, create a curve plot that completes as the movie advances in time.

> You will need: A 2D or 3D time-varying dataset and a Curve file containing the curve that will be animated

Reviewer: ```suggestion
You will need: A 2D or 3D time-varying dataset and a Curve file containing the curve that will be animated.
```

If you use a 3D dataset, you will want to make sure that you set the view to something appropriate and that you position the 3D plot in the upper portion of the vis window.
The 2D curve plot will be composited into the bottom part of the window.

.. _curveoverlay_preview:

.. figure:: images/curveoverlay_preview.png 
   :scale: 75%
   
   Curve Overlay preview.

> :numref:`Figure %s<curveoverlay_options>` shows the options for the 2D/3D plots:
> * Select your time varying database.
> * Select the plot and variable that you want to plot.
> * Adjust title and colors.

Reviewer: ```suggestion
:numref:`Figure %s<curveoverlay_options>` shows the options for the 2D/3D plots:

* Select your time varying database.
* Select the plot and variable that you want to plot.
* Adjust title and colors.
```

Reviewer: the bullets are not displaying properly

.. _SidebySideMovie:

Side by Side Movie
^^^^^^^^^^^^^^^^^^
To make a side by side movie you can use the **Create template** option.
Before saving the movie with this template, you must set up the plots that you want to make in multiple vis windows, as shown in :numref:`Figure %s<sidebyside_plots>`.

.. _sidebyside_plots:

.. figure:: images/sidebyside_plots.png 
   :scale: 75%
   
   Setting up plots for Side by Side movie generation.

Since you are creating a new movie template, you first must set up the movie viewports.
You can create many viewports and the viewports can be adjacent or can overlap (e.g. picture in picture).
When viewports overlap, you can make the top viewports semi-transparent, make a background color transparent, or even add drop shadows.
There are some existing layouts if you don't want to get fancy.
Select the **Side by Side** layout and VisIt will create 2 viewports next to one another.

.. _sidebyside_layout_selection:

.. figure:: images/sidebyside_layout_selection.png 
   :scale: 75%
   
   Selecting Side by Side layout.

.. _sidebyside_viewports:

.. figure:: images/sidebyside_viewports.png 
   :scale: 75%
   
   The Side by Side viewports.

> After you create movie viewports, you must map sequences of images to the viewports.
> Sequences are series of images and they can be produced from the plots in your vis windows or transitions such as fades, etc.

Reviewer: ```suggestion
After you create movie viewports, you must map sequences of images to the viewports.
These can be produced from the plots in your vis windows or transitions such as fades, etc.
```

I thought it was confusing to say that sequences of images are series of images.

For a side by side movie, you will want to map the image sequence produced from vis window 1 to movie viewport 1.
You will also map the image sequence from vis window 2 to movie viewport 2.
If you had more vis windows, you could map more than one vis window to a movie viewport.

> That has the effect that all images from a particular vis window are used as inputs for the movie viewport and then the next vis window that was mapped to the movie viewport, and so on.

Reviewer: This sentence confused me. Do you mean that ensuing vis windows will be used as inputs for the movie viewport as well?

Author: Confusing to me as well, I grabbed it from Side by Side movie page on visitusers. Should I just remove that sentence?

Reviewer: I'm reading it again and the more I read it, the less I understand what it is trying to say. If you have some idea of what it is supposed to say then I would advocate for replacing it, but as it stands it doesn't seem to be adding anything. Removing it seems like a decent idea.

Author: How about (I just tried this to see what would happen):

When more than one vis window is mapped to the same viewport, the movie will contain the animation from the first vis window followed by the animation from the second vis window.

Reviewer: That sounds good.
